File: DSA/Day15/ques2.cpp
#include<iostream>
 #include<queue>
 using namespace std;

 class Node{
    public:
    int data;
    Node* left;
    Node* right;

    Node(int val){
        this->data=val;
        this->left=NULL;
        this->right=NULL;
    }
 };

 Node* createtree(){
    int data;
    cout<<"Enter Value: "<<endl;
    cin>>data;

    if(data==-1){
        return NULL;
    }

    Node* root=new Node(data);

    cout<<"left child of "<<root->data<<endl;
    root->left=createtree();

    cout<<"right child of "<<root->data<<endl;
    root->right=createtree();
 }

 void levelorder(Node* root){
    queue<Node*>q;
    q.push(root);
    q.push(NULL);
  
    while(q.size()>1){
        Node* front=q.front();
        q.pop();

        if(front==NULL){
            cout<<endl;
            q.push(NULL); 
        }

        else{
            cout<<front->data<<" ";

            if(front->left!=NULL){
               q.push(front->left);
            }
            if(front->right!=NULL){
                q.push(front->right);
            }
        }
    }
 }
 int main(){
    Node* root=createtree();

    cout<<"Level-order trversal tree"<<endl;
    levelorder(root);
 }

File: DSA/Asmi_Garg/DAY 17/ques2.java
class TreeNode {
    int val;
    TreeNode left;
    TreeNode right;
    
    TreeNode(int x) {
        val = x;
    }
}

public class BST {
    public TreeNode insertIntoBST(TreeNode root, int val) {
        if (root == null) {
            return new TreeNode(val);
        }
        if (val < root.val) {
            root.left = insertIntoBST(root.left, val);
        } else if (val > root.val) {
            root.right = insertIntoBST(root.right, val);
        }
        return root;
    }

    public static void main(String[] args) {
        BST tree = new BST();
        TreeNode root = new TreeNode(4);
        root.left = new TreeNode(2);
        root.right = new TreeNode(7);
        root.left.left = new TreeNode(1);
        root.left.right = new TreeNode(3);
        
        int valueToInsert = 5;
        
        TreeNode updatedRoot = tree.insertIntoBST(root, valueToInsert);
        printInOrder(updatedRoot);  // Helper function to print the BST in-order
    }

    public static void printInOrder(TreeNode node) {
        if (node != null) {
            printInOrder(node.left);
            System.out.print(node.val + " ");
            printInOrder(node.right);
        }
    }
}
